Add cmake toggle switch around arbor backend (#1535)
nest/nest-simulator: CMakeLists.txt

set( with-music OFF CACHE STRING "Request the use of MUSIC. Optionally give the directory, where MUSIC is installed. [default=OFF]" )
set( with-sionlib OFF CACHE STRING "Request the use of SIONlib. Optionally give the directory where SIONlib is installed. [default=OFF]" )
set( with-boost OFF CACHE STRING "Request compilation with Boost, required for C++ testsuite. Optionally give directory with Boost installation. [default=OFF]" )

> set( with-arborbackend OFF CACHE STRING "Request compilation with Arbor backend capability actived. Requires MPI,PYTHON & MPI4PY [default=OFF]" )

Reviewer: ```suggestion
set( with-recordingbackend-arbor OFF CACHE STRING "Request compilation of the recording backend for Arbor. Requires MPI, Python and mpi4py [default=OFF]" )
```

nest_process_with_sionlib()
nest_process_with_mpi4py()
nest_process_with_boost()

> nest_process_with_arbor_backend()

Reviewer: ```suggestion
nest_process_with_recordingbackend_arbor()
```
